src/main/java/seedu/duke/data/timetable/Event.java

```diff
@@ -217,6 +219,96 @@ public String toString() {
+        ArrayList<String> results = new ArrayList<>();
+        events.sort(Comparator.comparing(Event::getDate).thenComparing(Event::getTime));
+        int i = 1;
+        for (Event event : events) {
+            ArrayList<String> eventString = event.toStringArray();
+            eventString.set(0, i++ + ". " + eventString.get(0));
+            results.addAll(eventString);
+            results.add(" ");
+        }
+        results.remove(results.size() - 1);
+        return results;
+    }
+
+    /**
+     * Converts all events in the ArrayList provided into stringArray format, appends them together and returns it.
+     *
+     * @param events Events in the same time period.
+     * @return Non-index ArrayList of strings to represent all events in this time period.
+     */
+    public static ArrayList<String> calendarToStringNoIndex(ArrayList<Event> events) {
+        ArrayList<String> results = new ArrayList<>();
+        events.sort(Comparator.comparing(Event::getDate).thenComparing(Event::getTime));
+        for (Event event : events) {
+            ArrayList<String> eventString = event.toStringArray();
+            results.addAll(eventString);
+            results.add(" ");
+        }
+        return results;
+    }
+
+    /**
+     * Converts a HashMap calendar to an ArrayList of strings representing all events in the calendar.
+     * Provides a day wrapper around the calendarToStringNoIndex.
+     *
+     * @param calendar Calendar storing events to print.
+     * @return ArrayList of Strings to be sequentially formatted in the formatter.
+     */
+    public static ArrayList<String> monthCalendarToString(HashMap<Integer, ArrayList<Event>> calendar) {
+        ArrayList<String> results = new ArrayList<>();
+        ArrayList<Integer> days = new ArrayList<>(calendar.keySet());
+        Collections.sort(days);
+        for (Integer day : days) {
+            results.addAll(calendarToStringNoIndex(calendar.get(day)));
+        }
+        return results;
+    }
+
+    /**
+     * Converts a HashMap calendar to an ArrayList of strings representing all events in the calendar.
+     * Provides a month comparison around monthCalendarToString.
+     *
+     * @param calendar Calendar storing events to print.
+     * @return ArrayList of Strings to be sequentially formatted in the formatter.
+     */
+    public static ArrayList<String> yearCalendarToString(HashMap<Month, HashMap<Integer, ArrayList<Event>>> calendar) {
+        ArrayList<String> results = new ArrayList<>();
+        ArrayList<Month> months = new ArrayList<>(calendar.keySet());
+        months.sort(Month::compareTo);
+        for (Month month: months) {
+            results.add(month.name());
+            results.addAll(monthCalendarToString(calendar.get(month)));
+            results.add(" ");
+        }
+        results.remove(results.size() - 1);
+        results.remove(results.size() - 1);
+        return results;
+    }
+
```

Comment:
These functions can just sort and compare the events and eventually just return an arrayList of event.